### PR TITLE
feat: add async event iterators

### LINKED ISSUE #879: Async Assistant Streaming Helpers
### Confirm this is a feature request for the Node library and not the underlying OpenAI API.

- [X] This is a feature request for the Node library

### Describe the feature or improvement you're requesting

When using the [Assistant Streaming Helpers](https://github.com/openai/openai-node/blob/master/helpers.md#helpers) it is sometimes desirable to have an async/await level of control for the event listener. For example, an `imageFileDone` listener that would move a file to a new destination as part of tools being submitted.

### Additional context

I am the author of Experts.js (https://github.com/metaskills/experts) and I felt this particular pain point in a few of my projects. I've not heard of other's needs for this behavior. That said, I thought that would be helpful for more context.

https://github.com/metaskills/experts?tab=readme-ov-file#streaming--events

Thanks y'all.

### PR DESCRIPTION
**Pull request by HAYDEN-OAI**
## Summary

- add a typed `events(eventName)` async iterator to `EventStream`
- document awaiting repeated `imageFileDone` assistant helper events
- add coverage for ordering, queued events, and stream failures

## Why

Issue #879 needs async/await control for SDK convenience events such as `imageFileDone`. Existing `.on()` callbacks are synchronous, so callers cannot naturally process each repeated event sequentially.

## Behavior

`stream.events(eventName)` yields the emitted argument tuple for every occurrence. It queues events while the consumer awaits work, ends with the stream, and surfaces stream errors or aborts after previously queued events have been drained. Existing `.on()` behavior is unchanged.

## Validation

- `pnpm exec jest tests/lib/EventStream.test.ts --runInBand`
- `pnpm exec tsc --noEmit`
- `pnpm run lint`

Resolves #879

### CODE REVIEW

File: src/lib/EventStream.ts [hunk 1]
```diff
@@ -157,6 +157,110 @@ export class EventStream<EventTypes extends BaseEvents> {
...
+      ended = true;
+      cleanup();
+      if (!pushQueue.length) {
+        rejectReader();
+        finishReaders();
+      }
+    };
+
+    if (!ended) {
+      this.#catchingPromiseCreated = true;
```
[the remark this answers is not in the document]
Comment (HAYDEN-OAI) [reply src/lib/EventStream.ts]: good catch, thanks! the iterator no longer mutates `#catchingPromiseCreated`. while it is active, its error and abort listeners still handle failures; once `return()` cleans those listeners up, a later unhandled stream failure is reported normally again. i also added a regression test for that cleanup path in e7cd5be2.

**Review by jbeckwith-oai — APPROVED** [review]

**Review by jbeckwith-oai — APPROVED** [review]
@-

**Comment by jbeckwith-oai** [comment]
Comprehensive review completed against e7cd5be2.

Original issue reproduced on base 95b54e58: a minimal EventStream async imageFileDone listener received two events while the first handler was blocked, producing start:first, start:second, end:second. This confirms #879: synchronous callbacks do not serialize awaited work. The reproduction passed with node_modules/.bin/jest tests/lib/EventStream.issue879-repro.test.ts --runInBand (1/1).

Validation:
- Authored tests plus six review edge cases passed (10/10): queued-before-read delivery, independent subscribers, multiple pending reads on end/failure, direct error/abort iteration, zero-argument tuples, cancellation without stream abort, and creation after end.
- Scoped TypeScript validation passed for the implementation and review cases.
- Prettier and ESLint passed on changed files.
- CI is green for build, lint, Node 20/22/24/26.2 tests, ecosystem tests, breaking-change detection, Agents SDK regressions, CodeQL, and the 237-test SDK integration run.

Compatibility:
- Additive API; existing on, once, emitted, and AssistantStream iteration behavior is unchanged.
- Event-name generics preserve exact argument tuples.
- Events remain ordered during awaited consumer work; terminal failures follow queued events; explicitly iterating error or abort yields those events.
- Iterator return removes only its own listeners and does not abort the stream. The follow-up commit restores normal unhandled-failure behavior after cleanup.
- No in-repository subclass has an events member collision. I found no runtime, type, or API breaking change.

No blocking or nonblocking findings. Approved.

**Review by jbeckwith-oai — APPROVED** [review]
Approved after reproducing issue #879 on base 95b54e58 and validating e7cd5be2 with 10 targeted tests plus scoped type, lint, formatting, CI, compatibility, and edge-case review. No blocking or nonblocking findings. Full evidence: https://github.com/openai/openai-node/pull/1977#issuecomment-4937963235
